nullaway/src/main/java/com/uber/nullaway/generics/GenericsChecks.java

```diff
@@ -1035,11 +1035,27 @@ public Nullness getGenericParameterNullnessAtInvocation(
       }
     }
 
-    if (!(tree.getMethodSelect() instanceof MemberSelectTree) || invokedMethodSymbol.isStatic()) {
+    if (tree instanceof MethodInvocationTree) {
+      if (!(((MethodInvocationTree) tree).getMethodSelect() instanceof MemberSelectTree)
+          || invokedMethodSymbol.isStatic()) {
+        return Nullness.NONNULL;
+      }
+    } else if (!(tree instanceof NewClassTree) || invokedMethodSymbol.isStatic()) {
       return Nullness.NONNULL;
     }
-    Type enclosingType =
-        getTreeType(((MemberSelectTree) tree.getMethodSelect()).getExpression(), config);
+
+    Type enclosingType = null;
+    if (tree instanceof MethodInvocationTree) {
+      enclosingType =
+          getTreeType(
+              ((MemberSelectTree) ((MethodInvocationTree) tree).getMethodSelect()).getExpression(),
+              config);
+
+    } else if (tree instanceof NewClassTree) {
+      NewClassTree newClassTree = (NewClassTree) tree;
+      enclosingType = getTreeType(newClassTree, config);
```

Comment:
Add a comment explicitly indicating we get the type arguments used to check the constructor call from the call itself

Reply:
Not addressed